modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -451,6 +470,138 @@ TensorMechanicsAction::actOutputGeneration()
+      auto name = mat_params.get<std::string>("eigenstrain_name");
+
+      // Check for base_name prefix
+      if (mat_params.isParamValid("base_name"))
+        name = mat_params.get<std::string>("base_name") + '_' + name;
+
+      // Check block restrictions
+      if (!blk)
+        mooseError("Internal error, Material object that does not inherit form BlockRestricted");
+      const std::set<SubdomainID> & blocks =
+          blk->blockRestricted() ? blk->blockIDs() : blk->meshBlockIDs();
+
+      if (std::includes(blocks.begin(), blocks.end(), _subdomain_ids.begin(), _subdomain_ids.end()))
+      {
+        material_eigenstrain_map[name].insert(blocks.begin(), blocks.end());
+        eigenstrain_set.insert(name);
+      }
+    }
+
+    // Account for reduced eigenstrains and CompositeEigenstrains
+    if (mat_name == "ComputeReducedOrderEigenstrain")
+    {
+      auto input_eigenstrain_names =
+          mat_params.get<std::vector<MaterialPropertyName>>("input_eigenstrain_names");
+      remove_reduced_set.insert(input_eigenstrain_names.begin(), input_eigenstrain_names.end());
+    }
+    // Account for CompositeEigenstrains
+    if (mat_name == "CompositeEigenstrain")
+    {
+      auto remove_list = mat_params.get<std::vector<MaterialPropertyName>>("tensors");
+      for (auto i : remove_list)
+        remove_reduced_set.insert(i);
+    }
+
+    // Account for MaterialConverter , add or remove later
+    if (mat_name == "RankTwoTensorMaterialConverter")
+    {
+      std::vector<std::string> remove_list;
+      std::vector<std::string> add_list;
+
+      if (mat_params.isParamValid("ad_props_out") && mat_params.isParamValid("reg_props_in") &&
+          _use_ad)
+      {
+        remove_list = mat_params.get<std::vector<std::string>>("reg_props_in");
+        add_list = mat_params.get<std::vector<std::string>>("ad_props_out");
+      }
+      if (mat_params.isParamValid("ad_props_in") && mat_params.isParamValid("reg_props_out") &&
+          !_use_ad)
+      {
+        remove_list = mat_params.get<std::vector<std::string>>("ad_props_in");
+        add_list = mat_params.get<std::vector<std::string>>("reg_props_out");
+      }
+
+      // These vectors are the same size as checked in MaterialConverter
+      for (unsigned int index = 0; index < remove_list.size(); index++)
+        remove_add_map.emplace(remove_list[index], add_list[index]);
+    }
+  }
+  // All the materials have been accounted for, now remove or add parts
+
+  // Remove names which aren't eigenstrains (converter properties)
+  for (auto remove_add_index : remove_add_map)
+  {
+    const bool is_in = eigenstrain_set.find(remove_add_index.first) != eigenstrain_set.end();
+    if (is_in)
+    {
+      eigenstrain_set.erase(remove_add_index.first);
+      eigenstrain_set.insert(remove_add_index.second);
+    }
+  }
+  for (auto index : remove_reduced_set)
+  {
+    if (eigenstrain_set.find(index) != eigenstrain_set.end())
+      eigenstrain_set.erase(index);
```

Comment:
do you need to check if it is in before you erase? (no you don't, just looked it up)